File: video_router/adapters/hailuo20.py
"""
hailuo20.py — Hailuo 2.0 adaptoru (HAZIR — model 23210)
=======================================================
Hailuo pipeline: both + end_only sahneler buraya gelir.
Ortak akis _hailuo.submit (hailuo_pipeline.py yeniden kullanilir; OSS upload + yy imza +
frame_mode + result-poll + OTOMATIK indirme).

Model ID: 23210 (mevcut hailuo_pipeline'in sabiti). "2.0" adi teyit edilmedi ama
video inip kaliteliyse sorun degil (karar B). Hata/kotu kalite olursa 2.0 curl'u yakalanir.
Token/cookie/project: VIDEO klasorunde (hailuo_token.txt / hailuo_cookie.txt / hailuo_project.txt).
"""

from .. import core
from . import _hailuo

MODEL_ID = "23210"


def _generate(job: core.Job):
    return _hailuo.submit(job, model_id=MODEL_ID, tag="Hailuo 2.0")


SPEC = core.register(core.AdapterSpec(
    key="hailuo2.0",
    provider="hailuo",
    model_tag="hailuo20",
    modes={"both", "end_only"},
    ready=True,
    generate=_generate,
    token_files=["hailuo_token.txt", "hailuo_cookie.txt", "hailuo_project.txt"],  # VIDEO klasorunde
    description="Hailuo 2.0 (both + end_only, model 23210) — HAZIR.",
))

File: src/lib/pipeline/router/adapters/hailuo20.ts
import { submitHailuoJob } from "../../hailuo";
import { HAILUO_MODEL_20 } from "../../hailuo/constants";
import { type Job, register } from "../core";

register({
  key: "hailuo2.0",
  provider: "hailuo",
  modelTag: "hailuo20",
  modes: new Set(["both", "end_only"]),
  ready: true,
  description: "Hailuo 2.0 (both + end_only, model 23210)",
  generate: async (job: Job) => {
    return submitHailuoJob({
      videoDir: job.videoDir,
      modelId: HAILUO_MODEL_20,
      label: String(job.scene.label || ""),
      frameMode: String(job.scene.frame_mode || "both"),
      prompt: job.prompt,
      startImage: job.startImage,
      endImage: job.endImage,
      outPath: job.outPath,
      resumeVidId: job.resumeVidId,
      duration: job.duration,
      resolution: job.resolution ?? undefined,
      promptOptimizer: job.promptOptimizer !== false,
      skipQueueGate: job.skipQueueGate,
      preGenerate: job.preGenerate,
      onSubmit: job.onSubmit,
    });
  },
});
